src/main/java/service/ReservationService.java

```diff
@@ -33,10 +33,10 @@ public String createReservation(String nameUser, String lastname, String nameBoo
         return "User or book not found";
     }
 
-    public ArrayList<Reservation> searchReservations(String name, String lastname) {
-        User user = userService.getUser(name, lastname);
+    public ArrayList<Reservation> searchReservations(int id) {
+        User user = userService.getUser(id);
         if (user != null) {
-            return reservationRepo.getUserReservations(userService.getUser(name, lastname));
+            return reservationRepo.getUserReservations(userService.getUser(id));
```

Comment:
The user is already referenced in line 37, its looking twice for the same user. Imagine using a real DB.